numpy/core/tests/test_array_coercion.py

```diff
@@ -141,12 +141,6 @@ class TestStringDiscovery:
             [object(), 1.2, 10**43, None, "string"],
```

Comment:
maybe its already tested somewhere, but if not, would be nice to have an array   
```
x = np.array(["qwwqwd", "qwdqwdwdqdqdwq"], dtype=object)
z = x.astype("S")
```
and make sure it finds the right dtype obj ?

similarly, since we are supporting it in the code:

```
 x = np.array(["2020-03-03", "2020-03-04"], dtype="S")
 z = x.astype(np.datetime64)
```

also a test for assigning np.datetime64 obj to an element of string array, to test the behavior before this PR.

Reply:
Hmmm, the datetime test seemed to cover these reasonably well, so I did not add a new one. I am pretty sure we already have tests for the `astype` case, at least simple ones. Probably in `test_api` or `test_multiarray`, but I can look it up later.

Reply:
did you find the test?

Reply:
ping @seberg, not sure if you are able to find this test already.

Reply:
Oh, here I guess: https://github.com/numpy/numpy/blob/2990d6e9102e4d7c3ff6cb1b69702c42bddaa738/numpy/core/tests/test_datetime.py#L646

Reply:
yes got it. i was thinking more about this one :
```
x = np.array(["qwwqwd", "qwdqwdwdqdqdwq"], dtype=object)
z = x.astype("S")
```

Reply:
You mean a test which has more than one nested elements?

Reply:
sorry for not being clear. i mean an array of string with dtype=object and after that astype called on the array. The test would be to make sure it finds the right dtype object which in this case is z.dtype == '|S14'.

Reply:
Ahh, I suppose there also tests here: https://github.com/numpy/numpy/blob/14d3173931cd958495bf592e0078213a6a66ab1d/numpy/core/tests/test_api.py#L238

but true, maybe including `.astype()` here is just as well, I did it in most of the other tests after all...

Reply:
cool ! thanks for the link, i think thats enough as long as its tested. I am fine either way w.r.t adding it to array coercion tests.